### PR TITLE
Second try of wildcard support for testresult files

### PR DESCRIPTION
**Pull request by Autom8edChaos**
Hi,

I messed up with my fork, so I recreated a branch from scratch. This is the new Pull request for the topic discussed in [this closed pull request](https://github.com/picklesdoc/pickles/pull/421).

### CODE REVIEW

File: src/Pickles/Pickles.Test/WhenParsingCommandLineArguments.cs [hunk 1]
```diff
@@ -465,6 +571,7 @@ public void ThenCanParseResultsFormatSpecrunWithShortFormSuccessfully()
         [Test]
         public void ThenCanFilterOutNonExistingTestResultFiles()
         {
+            FileSystem.AddFile(@"c:\results1.xml", "<xml />");
```
Comment (dirkrombauts) [inline src/Pickles/Pickles.Test/WhenParsingCommandLineArguments.cs]: Why do we need this line?

File: src/Pickles/Pickles.Test/WhenParsingCommandLineArguments.cs [hunk 2]
```diff
@@ -232,6 +232,112 @@ public void ThenCanParseResultsFileAsSemicolonSeparatedListThatEndsWithASemicolo
...
+            FileSystem.AddFile(@"c:\results2.xml", "<xml />");
+            var args = new[] { @"-link-results-file=c:\results*.xml" };
+
+            var configuration = new Configuration();
+            var commandLineArgumentParser = new CommandLineArgumentParser(FileSystem);
+            bool shouldContinue = commandLineArgumentParser.Parse(args, configuration, TextWriter.Null);
+
+            Check.That(shouldContinue).IsTrue();
+            Check.That(configuration.HasTestResults).IsTrue();
+            Check.That(configuration.TestResultsFiles.Count()).IsEqualTo(2);
```
Comment (dirkrombauts) [inline src/Pickles/Pickles.Test/WhenParsingCommandLineArguments.cs]: Thanks for the unit tests. I like your test cases.

Here's a way to write the assertions with more focus on their intention:

`Check.That(configuration.TestResultsFiles.Select(trf => trf.FullName)).ContainsExactly(@"c:\results1.xml", @"c:\results2.xml");`

Please change that here and in the other test cases as well.

**Comment by dirkrombauts** [comment]
By the way: would you be willing to also adapt the documentation? You can find it at https://github.com/picklesdoc/docs/blob/master/docs/ArgumentsTestResultsFile.md

**Comment by Autom8edChaos** [comment]
PR for docs here: https://github.com/picklesdoc/docs/pull/16

**Comment by dirkrombauts** [comment]
Thanks for the changes, they look good.

One question still stands: why do we need `FileSystem.AddFile(@"c:\results1.xml", "<xml />");` at line 574 of `src/Pickles/Pickles.Test/WhenParsingCommandLineArguments.cs`?

**Comment by Autom8edChaos** [comment]
@dirkrombauts You asked "One question still stands: why do we need FileSystem.AddFile(@"c:\results1.xml", "<xml />"); at line 574 of src/Pickles/Pickles.Test/WhenParsingCommandLineArguments.cs?", My previous answer got lost, hope this clarifies it:

For wildcard support, we split up the directory path and the filename that could be a pattern. Therefore pickles throws an error if a path does not exist. No error is thrown if the file does not exist. In this particular testcase, it is tested that no error is thrown when a testresults file does not exist, that is why we need the directory to exist where the non existing file is located (that sounds like Schrödingers file haha).

We have to make a choice: do we ignore it when a folder does not exist and just go on, or do we throw an exception on a not existing folder path. I prefer the latter since silent continue (aka Pokémon Exception Handling) is the root of all evil bugs.

**Comment by dirkrombauts** [comment]
Thanks for your contribution!

**Comment by dirkrombauts** [comment]
Released in version [2.14.0](https://github.com/picklesdoc/pickles/releases/tag/v2.14.0).

**Pull request merged**